walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/resolvers/OpenIdURLRequestResolver.kt

```diff
@@ -32,7 +37,41 @@ internal class OpenIdURLRequestResolver(val libraryConfiguration: LibraryConfigu
             "Provided VerifiedIdRequestInput is not supported."
         )
         if (libraryConfiguration.isPreviewFeatureEnabled(PreviewFeatureFlags.FEATURE_FLAG_OPENID4VCI_ACCESS_TOKEN))
-            TODO("Not implemented")
+            return resolveOpenId4VCIRequest(verifiedIdRequestInput)
         return OpenIdResolver.getRequest(verifiedIdRequestInput.url.toString())
     }
+
+    private suspend fun resolveOpenId4VCIRequest(verifiedIdRequestInput: VerifiedIdRequestURL): Any {
+        val requestUri = getRequestUri(verifiedIdRequestInput.url)
+            ?: throw RequestURIMissingException("Request URI is not provided in ${verifiedIdRequestInput.url}.")
+        fetchOpenID4VCIRequest(requestUri)
+            .onSuccess { requestPayload ->
+                return try {
+                    JSONObject(requestPayload.decodeToString())
+                    requestPayload.decodeToString()
+                } catch (e: Exception) {
+                    val presentationRequestContent =
+                        libraryConfiguration.serializer.decodeFromString(
+                            PresentationRequestContent.serializer(),
+                            requestPayload.toString()
+                        )
+                    OpenIdResolver.validateRequest(presentationRequestContent)
+                }
+            }
+            .onFailure { throw RequestURIMissingException("Request fetch failed because of ${it.message}.") }
```

Comment:
is this a request uri missing exception or a different one? Is there a way to have error nesting here so that we know which network request failed?

Reply:
I have created a class OpenId4VciRequestException in the future PR to handle all the error scenarios (Fetch errors and validation errors).